src/Forms/MoneyField.php

```diff
@@ -211,15 +293,27 @@ public function getAllowedCurrencies()
         return $this->allowedCurrencies;
     }
 
+    /**
+     * Assign locale to format this currency in
+     *
+     * @param string $locale
+     * @return $this
+     */
     public function setLocale($locale)
     {
-        $this->_locale = $locale;
+        $this->fieldAmount->setLocale($locale);
```

Comment:
I've modified this so that fieldAmount remains a NumericField, but is assigned setReadonly(true) instead of being type-shifted.